mux/src/tmux_commands.rs

```diff
@@ -0,0 +1,304 @@
+                .parse()?;
+            let pane_height = fields
+                .next()
+                .ok_or_else(|| anyhow!("missing pane_height"))?
+                .parse()?;
+            let pane_left = fields
+                .next()
+                .ok_or_else(|| anyhow!("missing pane_left"))?
+                .parse()?;
+            let pane_top = fields
+                .next()
+                .ok_or_else(|| anyhow!("missing pane_top"))?
+                .parse()?;
+
+            // These ids all have various sigils such as `$`, `%`, `@`,
+            // so skip those prior to parsing them
+            let session_id = session_id[1..].parse()?;
+            let window_id = window_id[1..].parse()?;
+            let pane_id = pane_id[1..].parse()?;
+
+            items.push(PaneItem {
+                session_id,
+                window_id,
+                pane_id,
+                pane_index,
+                cursor_x,
+                cursor_y,
+                pane_width,
+                pane_height,
+                pane_left,
+                pane_top,
+            });
+        }
+
+        log::info!("panes in domain_id {}: {:?}", domain_id, items);
+        let mux = Mux::get().expect("to be called on main thread");
+        if let Some(domain) = mux.get_domain(domain_id) {
+            if let Some(tmux_domain) = domain.downcast_ref::<TmuxDomain>() {
+                return tmux_domain.inner.sync_pane_state(&items);
+            }
+        }
+        anyhow::bail!("Tmux domain lost");
+    }
+}
+
+#[derive(Debug)]
+pub(crate) struct CapturePane(TmuxPaneId);
+impl TmuxCommand for CapturePane {
+    fn get_command(&self) -> String {
+        format!("capturep -p -t {} -e -C\n", self.0)
+    }
+
+    fn process_result(&self, domain_id: DomainId, result: &Guarded) -> anyhow::Result<()> {
+        let mux = Mux::get().expect("to be called on main thread");
+        let domain = match mux.get_domain(domain_id) {
+            Some(d) => d,
+            None => anyhow::bail!("Tmux domain lost"),
+        };
+        let tmux_domain = match domain.downcast_ref::<TmuxDomain>() {
+            Some(t) => t,
+            None => anyhow::bail!("Tmux domain lost"),
+        };
+
+        let pane_map = tmux_domain.inner.remote_panes.borrow();
+        if let Some(pane) = pane_map.get(&self.0) {
+            let lock = pane.lock().unwrap();
+            lock.tx.send(result.output.to_owned()).unwrap();
+        }
+
+        Ok(())
+    }
+}
+
+#[derive(Debug)]
+pub(crate) struct SendKeys {
+    pub keys: Vec<u8>,
+    pub pane: TmuxPaneId,
+}
+impl TmuxCommand for SendKeys {
+    fn get_command(&self) -> String {
+        let mut s = String::new();
+        for &byte in self.keys.iter() {
+            write!(&mut s, "0x{:X}\r", byte).expect("unable to write key");
+        }
+        format!("send-keys -t {} {}", self.pane, s)
+        // FIXME: An unexpected duplicated command will prompt next line, why?
```

Comment:
After some search, I found this really hard to solve. Currently Tmux use `ESC k` to set the window name, which we don't support yet. It's really annoying to live with that window title echoed after every command..

Relate issue in mosh: https://github.com/mobile-shell/mosh/issues/992

Reply:
That `ESC k TITLE ST` sequence is pretty awkward because it doesn't match the typical OSC style escapes that use ST (string terminator).  That makes it difficult to fit into the parser at the right layer :-(

Do you know what is emitting that sequence?
http://manpages.ubuntu.com/manpages/precise/en/man1/tmux.1.html#names%20and%20titles
suggests that automatic-rename may be a factor: can we tell tmux to disable that option when we set up control mode?

Reply:
I found that it was my zsh. Set `DISABLE_AUTO_TITLE="true"` disable this sequence.